### ISSUE TITLE
[CoE Starter Kit - QUESTION] Is it okay to use PowerShell to obtain older Audit Log data and then import it into the Audit Log table

### ISSUE THREAD

**Issue by firefox-edge123** [microsoft/coe-starter-kit#7257]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

Regarding importing of older Audit Log data:
https://learn.microsoft.com/en-us/power-platform/guidance/coe/setup-auditlog-http#how-to-get-older-data

The Microsoft Purview portal seems to only allow you to select Start Times and End Times by 30 minute increments.
(see screenshot below)

It seems that Audit Log data can be obtained from PowerShell and exported into a .CSV file.
And this method allows you to select Start Times and End Times by the minute (e.g. 11:23 AM)
https://learn.microsoft.com/en-us/powershell/module/exchange/search-unifiedauditlog?view=exchange-ps

For example, I was able to obtain a CSV file of old Audit Log data using these 3 PowerShell command:
(this was only a test)

Import-Module ExchangeOnlineManagement

Connect-ExchangeOnline

Search-UnifiedAuditLog -StartDate "12/6/2023 11:52 PM" -EndDate "12/7/2023 11:59 PM" -RecordType PowerAppsApp | Export-CSV -Encoding UTF8 -NoTypeInformation "C:\temp\powerapp-launches.csv"

I compared the 2 exported CSV files, and the headers are similar, but with some differences.

The headers of the Microsoft Purview CSV file are:

- RecordId	
- CreationDate	
- RecordType	
- Operation	
- UserId	
- AuditData	
- AssociatedAdminUnits	
- AssociatedAdminUnitsNames

The headers of the PowerShell CSV file are:

- RecordType	
- CreationDate	
- UserIds	
- Operations	
- AuditData	
- ResultIndex	
- ResultCount	
- Identity	
- IsValid	
- ObjectState

I am just wondering, is there any reason why I should Not be importing .CSV files that are obtained with the  PowerShell method I mentioned above?

![purview](https://github.com/microsoft/coe-starter-kit/assets/128085471/72e4e36e-a1db-4150-a28f-6a03902c1d5e)

### What solution are you experiencing the issue with?

None

### What method are you using to get inventory and telemetry?

None

AB#1733

**Comment by Jenefer-Monroe** [comment]
I'm really not sure as I havent done this before. Give it a try and let us know if it works! Others will likely se this write up and benefit from your work!

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team

**Comment by firefox-edge123** [comment]
(I realized I forgot to post back here. The results of my testing is below)


It seems the "alternate Powershell method" did Not work properly.
If I had opened a PowerApp multiple times, the Audit log entries would appear in the CSV spreadsheet that was extracted by Powershell, but after doing Step 4, Step 5 and Step 6, it would only show one PowerApp launch entry, rather than multiple entries in the audit logs.

I ended up just following the instructions in the documentation to get the missing Audit log data from Microsoft Purview
https://learn.microsoft.com/en-us/power-platform/guidance/coe/setup-auditlog-http#how-to-get-older-data

After I downloaded the exported CSV file, I sorted the data by creation date (selected the entire table, data tab, sort, sort by CreationDate, My data has headers, OK.)
I then manually removed Audit Log entries that were already in the COE environment, then completed step 4, step 5, and step 6 of the documentation.
